### PR TITLE
feat: add custom annotations to kratos courier

### PR DESCRIPTION
**Pull request by sthh**
It is currently not possible to add annotations to the kratos courier pod.
This PR adds annotations to the kratos courier statefulset spec template. Following the current behavior, it reuses the `deployment.annotations` config, which is also used  to annotate the kratos deployment and spec as well as the courier statefulset itself.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- ~I have referenced an issue containing the design document if my change
      introduces a new feature.~ No Issues was opened, since I the PR was already implemented for our use case.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- ~I have added tests that prove my fix is effective or that my feature
      works.~ Not applicable, since the templates aren't tested
- [x] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

**Review by Demonsthere — CHANGES_REQUESTED** [review]
Hello there, and sorry it took so long 😓 
Could you please update the PR, as there were changes in kratos since you've opened this. 
Right now, we would prefer to have a separate value for statefulset and deployment like here https://github.com/ory/k8s/blob/master/helm/charts/kratos/values.yaml#L304

**Comment by sthh** [comment]
Hi Jakub,
thanks for the feedback. I've rebased my branch on the current master.
While I would also prefer to use separate values for the statefulset, please note that `Values.deployment` is currently used for most of the configuration in `statefulset-mail.yaml`, including the annotations for the statefulset (I've only added it to the spec). Would you prefer a separate value anyways?

**Comment by Demonsthere** [comment]
Hi there, 
I have been trying to introduce the model of ternary values, which could be used here:

```yaml
{{- $annotations := ternary .Values.statefulSet.annotations .Values.deployment.annotations (not (empty .Values.statefulSet.annotations )) -}}
```

This would allow us to use the dedicated value if it exist, and fallback to deployment if not :)

**Comment by sthh** [comment]
Sounds good! I will update the PR next week and change all relevant config values in the statefulset to use ternary values.

File: helm/charts/kratos/values.yaml [hunk 1]
```diff
@@ -313,6 +313,41 @@ statefulSet:
     #    cpu: 100m
     #  memory: 128Mi
 
+    extraEnv: []
+      # -- If you want to mount external volume
+      # For example, mount a secret containing Certificate root CA to verify database
+      # TLS connection.
+    extraVolumes: []
```
Comment (Demonsthere) [inline helm/charts/kratos/values.yaml:320]:
```suggestion
    # -- If you want to mount external volume
    # For example, mount a secret containing Certificate root CA to verify database
    # TLS connection.
    extraVolumes: []
```

File: helm/charts/kratos/values.yaml [hunk 2]
```diff
@@ -313,6 +313,41 @@ statefulSet:
...
+    extraInitContainers: {}
+    # extraInitContainers: |
+    #  - name: ...
+    #    image: ...
+
+    # -- If you want to add extra sidecar containers.
+    extraContainers: {}
+      # extraContainers: |
+      #  - name: ...
+      #    image: ...
```
Comment (Demonsthere) [inline helm/charts/kratos/values.yaml:339]:
```suggestion
    extraContainers: {}
    # extraContainers: |
    #  - name: ...
    #    image: ...
```

**Review by Demonsthere — APPROVED** [review]
Great job! Only found some minor indentation issues :)

**Pull request merged**